src/irr/video/COpenGLSpecializedShader.h

```diff
@@ -47,6 +76,8 @@ class COpenGLSpecializedShader : public IGPUSpecializedShader
     core::smart_refctd_ptr<asset::CIntrospectionData> m_introspectionData = nullptr;
     SProgramBinary m_binary;
 
+    std::mutex m_getGLnameMutex;
```

Comment:
this mutex can go to `COpenGLDriver`... oh wait it already has one when you're trying to get  the correct `SAuxContext`

Reply:
yea, but i dont have to whole `getThreadContext()` just to get one GL name from one specific shader. It's shader-local for a reason

Reply:
resolved